src/actions/receive-profile.js

```diff
@@ -708,44 +728,36 @@ class ColascedFunctionsUpdateDispatcher {
 
   _dispatchUpdate(dispatch) {
     const updates = this._updates;
-    this._updates = {};
+    this._updates = new Map();
     this._requestedUpdate = false;
-    dispatch(coalescedFunctionsUpdate(updates));
+    dispatch(coalescedSymbolicationStep(updates));
   }
 
-  doSymbolicationStep(
+  enqueueSingleSymbolicationStep(
     dispatch: Dispatch,
     threadIndex: ThreadIndex,
-    oldFuncToNewFuncMap: FuncToFuncMap,
     symbolicationStepInfo: SymbolicationStepInfo
   ) {
     this._scheduleUpdate(dispatch);
-    if (!this._updates[threadIndex]) {
-      this._updates[threadIndex] = {
-        oldFuncToNewFuncMap,
-        symbolicationSteps: [symbolicationStepInfo],
-      };
-    } else {
-      for (const [oldFunc, funcIndex] of oldFuncToNewFuncMap) {
-        this._updates[threadIndex].oldFuncToNewFuncMap.set(oldFunc, funcIndex);
-      }
-      this._updates[threadIndex].symbolicationSteps.push(symbolicationStepInfo);
+    let threadSteps = this._updates.get(threadIndex);
+    if (threadSteps === undefined) {
+      threadSteps = [];
+      this._updates.set(threadIndex, threadSteps);
     }
+    threadSteps.push(symbolicationStepInfo);
   }
 }
 
-const gCoalescedFunctionsUpdateDispatcher = new ColascedFunctionsUpdateDispatcher();
+const gCoalescedSymbolicationStepDispatcher = new ColascedSymbolicationStepDispatcher();
```

Comment:
Optional: It's a bit odd that this is named `gCoalescedFunctionsUpdateDispatcher`, as it's not a global, but a module-scoped variable. I would prefer `_coalescedFunctionsUpdateDispatcher` or `coalescedFunctionsUpdateDispatcher` as the name. Or maybe something like `coalescedFunctionsUpdateDispatcherSingleton`, although that's a mouthful!

Reply:
Renamed to `_symbolicationStepQueueSingleton`.